docs/src/appendix/symmetric_tutorial.md

```diff
@@ -0,0 +1,1291 @@
+implied internal structure is. Therefore, we require some more discussion before we can
+actually move on to an example.
+
+We'll start by discussing the general structure of a `TensorMap` which is symmetric under a
+non-Abelian group symmetry. We then given an example based on $\mathrm{SU}(2)$, where we
+construct the Heisenberg Hamiltonian using two different approaches. Finally, we show how
+the more intuitive approach can be used to obtain an elegant generalization to the
+$\mathrm{SU}(N)$-symmetric case.
+
+
+### Block sparsity revisited: the Wigner-Eckart theorem
+
+Let us recall some basics of representation theory first. Consider a group $G$ and a
+corresponding representation space $V$, such that every element $g \in G$ can be realized as
+a unitary operator $U_g : V \to V$. Let $h$ be a `TensorMap` whose domain and codomain are
+given by the tensor product of two of these representation spaces. By definition, the
+statement that '$h$ is symmetric under $G$' means that
+```@raw html
+<center><img src="../img/symmetric_tutorial/symmetric_tensor.svg" alt="symmetric_tensor" class="color-invertible" style="zoom: 170%"/></center>
+```
+for every $g \in G$. If we label the irreducible representations of $G$ by $l$, then any
+representation space can be decomposed into a direct sum of irreducible representations, $V
+= \bigoplus_l V^{(l)}$, in such a way that $U_g$ is block-diagonal where each matrix block
+is labeled by a particular irrep $l$. For each irrep space $V^{(l)}$ we can define an
+orthonormal basis labeled as $\ket{l, m}$, where the auxiliary label $m$ can take
+$\text{dim}\left( V^{(l)} \right)$ different values. Since we know that tensors are
+multilinear maps over tensor product spaces, it is natural to consider the tensor product of
+representation spaces in more detail.
+
+[From the representation theory of groups](https://en.wikipedia.org/wiki/Tensor_product_of_representations#Clebsch%E2%80%93Gordan_theory),
+it is known that the product of two irreps can in turn be decomposed into a direct sum of
+irreps, $V^{(l_1)} \otimes V^{(l_2)} \cong \bigoplus_{k} V^{(k)}$. The precise nature of
+this decomposition, also refered to as the *Clebsch-Gordan problem*, is given by the
+so-called *Clebsch-Gordan coefficients*, which we will denote as $C^{k}_{l_1,l_2}$. This set
+of coefficients, which can be interpreted as a $\text{dim}\left( V^{(l_1)} \right) \times
+\text{dim}\left( V^{(l_2)} \right) \times \text{dim}\left( V^{(k)} \right)$ array, that
+encodes how a basis state $\ket{k,n} \in V^{(k)}$ corresponding to some term in the direct
+sum can be decomposed into a linear combination of basis vectors $\ket{l_1,m_1} \otimes
+\ket{l_2,m_2}$ of the tensor product space:
+```math
+\begin{equation}
+\label{eq:cg_decomposition}
+\ket{k,n} = \sum_{m_1, m_2} \left( C^{k}_{l_1,l_2} \right)^{n}_{m_1, m_2} \ket{l_1,m_1} \otimes \ket{l_2,m_2}.
+\end{equation}
+```
```

Comment:
In this general discussion, it is kind of assumed that every irrep only appears once in the fusion output (`SimpleFusion`), or otherwise it is implied that `k` plays a different role than `l_1` and `l_2`, with the latter two really being irrep labels, while `k` a way to enumerate all irreps appearing in the fusion product. Maybe a `!!!note` box can be a solution?